berryio_connector.py

```diff
@@ -0,0 +1,377 @@
+            endpoint = BERRYIO_SET_MODE + pin + '/' + param.get('mode', '')
+        elif param.get('value', None) is not None:
+            endpoint = BERRYIO_SET_VALUE + pin + \
+                '/' + str(param.get('value', ''))
+
+        # Progress
+        self.save_progress(USING_BASE_URL.format(
+            base_url=self._base_url, api_uri=self._api_uri, endpoint=endpoint))
+
+        # Connectivity
+        self.save_progress(phantom.APP_PROG_CONNECTING_TO_ELLIPSES, self._host)
+
+        # Make the rest call, note that if we try for cached and its not there, it will automatically go to start a new analysis.
+        # unless specified start a new as above.
+        ret_val, response = self._make_rest_call(
+            endpoint, action_result, params='', retry=True)
+        self.debug_print('Ret_val: {}'.format(ret_val))
+        """
+        OK:
+        """
+        if ret_val:
+            if response.text.strip().lower().startswith('ok'):
+                response_data = {'raw': response.text}
+                # Set the summary and response data
+                action_result.add_data(response_data)
+                # action_result.set_summary({ 'total_hops': len(response_data)})
+
+                # Set the Status
+                return action_result.set_status(phantom.APP_SUCCESS)
+            else:
+                return phantom.APP_ERROR
+        else:
+            return phantom.APP_ERROR
+
+    def handle_action(self, param):
+        """Function that handles all the actions"""
+
+        # Get the action that we are supposed to carry out, set it in the connection result object
+        action = self.get_action_identifier()
+
+        # Intialize it to success
+        ret_val = phantom.APP_SUCCESS
+
+        # self.debug_print('DEBUG Action: {}'.format(action))
+        # Bunch if if..elif to process actions
+        if action == self.ACTION_ID_SET_GPIO_MODE:
+            ret_val = self._set_gpio_mode(param)
+        elif action == self.ACTION_ID_SET_GPIO_VALUE:
+            ret_val = self._set_gpio_mode(param)
+        elif action == self.ACTION_ID_GET_GPIO_STATUS:
+            ret_val = self._get_gpio_status(param)
+        elif action == self.ACTION_ID_TEST_ASSET_CONNECTIVITY:
+            ret_val = self._test_connectivity(param)
+
+        # self.debug_print('DEBUG: ret_val: {}'.format(ret_val))
+
+        return ret_val
+
+
+if __name__ == '__main__':
+    """ Code that is executed when run in standalone debug mode
+    for .e.g:
+    python2.7 ./berryio.py /tmp/berryio.json
+    """
+
+    # Imports
+    import sys
+
+    import pudb
+
+    # Breakpoint at runtime
+    pudb.set_trace()
+
+    # The first param is the input json file
+    with open(sys.argv[1]) as f:
+
+        # Load the input json file
+        in_json = f.read()
+        in_json = json.loads(in_json)
+        print(json.dumps(in_json, indent=' ' * 4))
+
+        # Create the connector class object
+        connector = BerryIOConnector()
```

Comment:
In the future if we plan to support this ourselves, we'd want to add our usual debug lines

Reply:
done